src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1356,38 +1356,48 @@ impl Cage {
 
         if let Some(wrappedfd) = fdtable.get(&fd) {
             let mut filedesc_enum = wrappedfd.write().unwrap();
-            
-            let filetype: i8;
-
-            let flags = match &mut *filedesc_enum {
-                Epoll(obj) => {filetype = 0; &mut obj.flags},
-                Pipe(obj) => {filetype = 1; &mut obj.flags},
-                Stream(obj) => {filetype = 2; &mut obj.flags},
-                Socket(obj) => {filetype = 3; &mut obj.flags},
-                File(obj) => {filetype = 4; &mut obj.flags},
-            };
 
             match request {
                 FIONBIO => {
                     let arg_result = interface::get_ioctl_int(ptrunion);
-                    //matching the tuple
-                    match (arg_result, filetype) {
+                    //matching the tuple and passing in filedesc_enum
+                    match (arg_result, &mut *filedesc_enum) {
                         (Err(arg_result), ..)=> {
                             return arg_result; //syscall_error
                         }
-                        (Ok(arg_result), 3) => {
-                            let arg: i32 = arg_result;
+                        (Ok(arg_result), Socket(ref mut sockfdobj)) => {
+                            let flags = &mut sockfdobj.flags;
+                            let mut arg: i32 = arg_result;
+
                             if arg == 0 { //clear non-blocking I/O
                                 *flags &= !O_NONBLOCK;
                             }
                             else { //set for non-blocking I/O
                                 *flags |= O_NONBLOCK;
                             }
+
+                            let sid = Self::getsockobjid(&mut *sockfdobj);
+                            let locksock = NET_METADATA.write().unwrap().socket_object_table.get(&sid).unwrap().clone();
+                            let sockobj = locksock.read().unwrap();
+
+                            let ioctlret: i32 = sockobj.ioctl(request, &mut arg as *mut i32);
+                            
+                            if ioctlret < 0 {
+                                match Errno::from_discriminant(interface::get_errno()) {
+                                    Ok(i) => {return syscall_error(i, "ioctl", "The libc call to ioctl failed!");},
+                                    Err(()) => panic!("Unknown errno value from ioctl returned!"),
+                                };
+                            }
+
                             0
                         }
                         _ => {syscall_error(Errno::ENOTTY, "ioctl", "The specified request does not apply to the kind of object that the file descriptor fd references.")}
                     }
                 }
+                FIOASYNC => { //not implemented
+                    println!("ioctl(FIOASYNC) is not implemented, and just returns 0.");
```

Comment:
We should probably only do this so its logs on verbose and not always println

Reply:
Updated to log only on verbose.